### ISSUE TITLE
logging: Improve logging and async debugging experience

### ISSUE THREAD

**Issue by justinrubek** [GlareDB/glaredb#108]
We are currently using the tracing crate for logging, but there is potential for adding on to this for development. Especially as the application gets more complex and uses increasing amounts of async code this is becoming more difficult. When working on #63 it felt like there were major slowdowns at times because of this. I decided to learn more about the tracing library and [immediately they offer a few suggestions](https://tokio.rs/tokio/topics/tracing):

- [Tokio console](https://github.com/tokio-rs/console)
  - basically an async debugger
  - [included in nixpkgs](https://search.nixos.org/packages?channel=unstable&show=tokio-console&from=0&size=50&sort=relevance&type=packages&query=tokio)
- https://opentelemetry.io/

I do not have experience with any of these, but they look like they could be very useful and we could probably pick them up fairly simply. Integrating these tools, learning how to use them, writing a quick-guide, and making a pass on tracing could help significantly
